From 4d24f6882273d68af22d0bd11dfbd86836c0db9b Mon Sep 17 00:00:00 2001
From: Edwin Peer <edwin.peer@netronome.com>
Date: Fri, 21 Sep 2018 17:46:30 -0700
Subject: [PATCH 4/9] [PCI.OUT] introduce NFD_PD_ALWAYS_HAS_CTM mode

This compile option unconditionally frees the CTM buffer if
NFD_PD_ALWAYS_HAS_CTM is defined. Packets sent to NFD from
CoreNIC always have an associated CTM buffer making this
check wasteful in a cycle constrained ME.

Signed-off-by: Edwin Peer <edwin.peer@netronome.com>
---
 me/blocks/vnic/pci_out_pd.uc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/me/blocks/vnic/pci_out_pd.uc b/me/blocks/vnic/pci_out_pd.uc
index 828283a0..84f6620b 100644
--- a/me/blocks/vnic/pci_out_pd.uc
+++ b/me/blocks/vnic/pci_out_pd.uc
@@ -792,7 +792,9 @@ add_wq_credits#:
     alu[$ticket, g_seq_mask, AND, io_work[SB_WQ_SEQ_wrd], >>SB_WQ_SEQ_shf]
 
     wsm_extract(isl, io_work, SB_WQ_CTM_ISL)
+#ifndef NFD_PD_ALWAYS_HAS_CTM
     beq[no_ctm_buffer#]
+#endif
 
     // Free CTM buffer
     alu[addr_hi, g_pkt_free_hi, OR, isl, <<24]
-- 
2.19.0

